src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +422,97 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+export class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [regionName: string]: SubtitleRegionContainer } = {};
+  private labelsInRegionCount: { [regionName: string]: number } = {};
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(private subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   */
+  removeLabel(labelToRemove: SubtitleLabel, event?: SubtitleCueEvent): void {
+    const region = event && event.region || 'default';
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[regionName].removeLabel(labelToRemove);
+    }
+    this.labelsInRegionCount[region]--;
+
+    // Remove container if no more labels are displayed
+    if (this.labelsInRegionCount[region] === 0) {
```

Comment:
If the `SubtitleRegionContainer` kept count of its labels, this could look like `subtitleRegionContainer.getLabelCount() === 0` or even `subtitleRegionContainer.isEmpty()`